doc/connection.rst

```diff
@@ -152,13 +172,26 @@ Connection Types
     :param timeout: Timeout for the connection in seconds. If a connection
                     cannot be established within this time, an error will be
                     raised.
-    :param password_prompt: A string with the password prompt used by
-                            ``sshpass``. Set this if your version of ``sshpass``
-                            uses something other than ``"[sudo] password"``.
+    :param platform: Specify the platform to be used. The generic :class:`~devlib.platform.Platform`
+                     class is used by default.
     :param sudo_cmd: Specify the format of the command used to grant sudo access.
-    :param options: A dictionary with extra ssh configuration options.
-
-
+    :param strict_host_check: Specify the ssh connection parameter ``StrictHostKeyChecking``,              
+    :param use_scp: Use SCP for file transfers, defaults to SFTP.
+    :param poll_transfers: Specify whether file transfers should be polled. Polling
+                           monitors the progress of file transfers and periodically
+                           checks whether they have stalled, attempting to cancel
+                           the transfers prematurely if so.
+    :param start_transfer_poll_delay: If transfers are polled, specify the length of
+                                      time after a transfer has started before polling
+                                      should start.
```

Comment:
If this parameter is set to 0 then the polling mechanism fails.

Reply:
Ah, the problem here is that I'm using the time `start_transfer_poll_delay` as a timeout when waiting for the transfer to start, when they aren't really the same. Should I expose this time as parameter, or just a constant, for the amount of time to wait for the transfer to start before cancelling?

Reply:
I don't think this would need to be exposed, is it expected to vary much / can a sensible constant be picked?

Reply:
I've edited the comment, as I no longer think it needs a timeout